src/components/inputs/multiButton/MultiButton.tsx

```diff
@@ -70,26 +70,36 @@ const MultiButton = React.forwardRef<HTMLButtonElement, MultiButtonProps>(functi
 		}
 	}, []);
 
+	const secondaryActionIcon = useMemo(() => {
+		if (icon) {
+			return icon;
+		}
+		if (dropdownOpen) {
+			return 'ChevronUpOutline';
+		}
+		return 'ChevronDownOutline';
+	}, [dropdownOpen, icon]);
+
 	const secondaryAction = useMemo<NonNullable<ButtonProps['secondaryAction']>>(
 		() => ({
-			icon,
-			onClick: openDropdown,
+			icon: secondaryActionIcon,
+			onClick: toggleDropdown,
 			disabled: disabledSecondary,
 			forceActive: dropdownOpen,
 			ref: secondaryButtonRef
 		}),
-		[disabledSecondary, dropdownOpen, icon, openDropdown]
+		[disabledSecondary, dropdownOpen, toggleDropdown, secondaryActionIcon]
 	);
 
 	return (
 		<StyledDropdown
+			{...dropdownProps}
 			items={items}
 			placement="bottom-end"
 			forceOpen={dropdownOpen}
 			onClose={closeDropdown}
 			disabled
 			disableRestoreFocus
```

Comment:
Imo all these props should be omitted from dropdownProps, since they will be always overwritten